File: backend/src/main/java/com/backend/api/user/dto/response/UserMyPageResponse.java
package com.backend.api.user.dto.response;

import com.backend.domain.user.entity.Role;
import com.backend.domain.user.entity.User;
import com.backend.domain.user.repository.UserRepository;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Builder
@Getter
@AllArgsConstructor
public class UserMyPageResponse {

    private Long userId;
    private String email;
    private String name;
    private String nickname;
    private int age;
    private String github;
    private String image;

    public static UserMyPageResponse fromEntity(User users){
        return UserMyPageResponse.builder()
                .userId(users.getId())
                .email(users.getEmail())
                .name(users.getName())
                .nickname(users.getNickname())
                .age(users.getAge())
                .github(users.getGithub())
                .image(users.getImage())
                .build();
    }


    @Getter
    public static class UserModify {
        private String email;
        private String password;
        private String name;
        private String nickname;
        private int age;
        private String github;
        private String image;
    }

    @Getter
    @Builder
    @AllArgsConstructor
    public static class SolvedProblem {
        private String title;              // 문제 제목
        private LocalDateTime modifyDate; // 수정일
    }
}

File: backend/src/main/java/com/backend/api/user/dto/response/UserMyPageResponse.kt
package com.backend.api.user.dto.response

import com.backend.domain.user.entity.User
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

data class UserMyPageResponse(
    @field:Schema(description = "유저 아이디", example = "5")
    val userId: Long,

    @field:Schema(description = "유저 이메일", example = "user@gmail.com")
    val email: String,

    @field:Schema(description = "유저 이름", example = "김아무개")
    val name: String,

    @field:Schema(description = "유저 닉네임", example = "김씨")
    val nickname: String,

    @field:Schema(description = "유저 나이", example = "25")
    val age: Int,

    @field:Schema(description = "유저 깃허브 주소", example = "github3")
    val github: String?,

    @field:Schema(description = "이미지 주소", example = "image3")
    val image: String?
) {

    class SolvedProblem(
        val title: String,// 문제 제목
        val modifyDate: LocalDateTime?// 수정일
    )

    companion object {
        fun fromEntity(user: User): UserMyPageResponse =
            UserMyPageResponse(
                userId = user.id,
                email = user.email,
                name = user.name,
                nickname = user.nickname,
                age = user.age,
                github = user.github,
                image = user.image
            )
    }
}